### PR TITLE
Shared widget

### PR DESCRIPTION
**Pull request by kghamilton89**
Refactor and migration of [Generic Widget](https://axibase.com/products/axibase-time-series-database/visualization/widgets/configuring-the-widgets/).

* I took the liberty of changing the name to Shared Widget Syntax since that is more similar to language we use in ATSD Docs. 
* There are several pages referred to (without links) which do not exist in `/charts` repo yet such as **Label Formatting**. It seems redundant to link to Legacy Docs and then re-link to new GH docs, so the plan is to go through and find all locations where links are needed and add them as the support pages are added.

### CODE REVIEW

File: .dictionary [hunk 1]
```diff
@@ -0,0 +1,6 @@
+TreeMap
```
Comment (rodionos) [inline .dictionary]: These look like names that are either functions or parameters. Also desired and undesired are not technical words, we are better off avoiding them and they probably belong to blacklist.

Comment (kghamilton89) [reply .dictionary]: TreeMap is an object class so I'll backtick it, and I'll remove the undesired incident of undesired.

Comment (kghamilton89) [reply .dictionary]: Though we do have a widget called TreeMap as well.

Comment (kghamilton89) [reply .dictionary]: https://axibase.com/products/axibase-time-series-database/visualization/widgets/treemap-widget/

Comment (rodionos) [reply .dictionary]: I think we can migrate to Treemap and treemap.

Comment (kghamilton89) [reply .dictionary]: All references to Treemap widget now use the non-Java class case.

**Pull request merged**